src/sql/azdata.proposed.d.ts

```diff
@@ -976,4 +976,168 @@ declare module 'azdata' {
 			expiresOn?: number
 		}
 	}
+
+	export enum DataProviderType {
+		TableDesignerProvider = 'TableDesignerProvider'
+	}
+
+	export namespace dataprotocol {
+		export function registerTableDesignerProvider(provider: designers.TableDesignerProvider): vscode.Disposable;
+	}
+
+	export namespace designers {
+
+		export function openTableDesigner(providerId: string, tableInfo: TableInfo): Promise<void>;
+
+		export interface TableDesignerProvider extends DataProvider {
+			getTableDesignerInfo(table: TableInfo): Promise<TableDesignerInfo>;
+			processTableEdit(table: TableInfo, data: DesignerData, edit: DesignerEdit): Promise<DesignerEditResult>;
+		}
+
+		/**
+		 * The information of the table.
+		 */
+		export interface TableInfo {
+			server: string;
+			database: string;
+			schema?: string;
+			name?: string;
+			isNewTable: boolean;
+			[key: string]: any;
+		}
+
+		export interface TableDesignerInfo {
+			view: TableDesignerView;
+			data: DesignerData;
+			columnTypes: string[];
+			schemas: string[];
+		}
+
+		export enum TableProperty {
+			Columns = 'columns',
+			Description = 'description',
+			Name = 'name',
+			Schema = 'schema',
+			Script = 'script'
+		}
+
+		export enum TableColumnProperty {
+			AllowNulls = 'allowNulls',
+			DefaultValue = 'defaultValue',
+			Length = 'length',
+			Name = 'name',
+			Type = 'type'
+		}
+
+		export interface TableDesignerView {
+			additionalTableProperties?: DesignerComponentType[];
```

Comment:
The name is properties but it's a list of component types? Shouldn't it be something like `additionalTablePropertyComponentTypes` or something?